webClient/src/app/shared/language-server/language-server.service.ts

```diff
@@ -16,15 +16,21 @@ export class LanguageServerService {
 
   config = { domain: 'ws://localhost:3000', endpoint: { hlasm: 'asmServer', json: 'jsonServer' } };
   connections: { name: string, connection: MessageConnection }[] = [];
+  enabled: boolean = true;
 
   constructor() { }
 
   getSettings(): any {
     return this.config;
   }
 
-  updateSettings(config: any) {
-    this.config = config;
+  getEnabled(): boolean {
+    return this.enabled;
+  }
+
+  updateSettings(result: any) {
+    this.config = JSON.parse(result.config);
```

Comment:
You should not do this without checks prior.
1 result is a bad name for an argument, but a good name for a return value.
2 you don't know that result.config exists, so if result=undefined, result.config=exception
3 you dont know that result is parse-able. if result.config = not a json, then json.parse(result.config)=exception
you should check for existence of the argument, then put a try {} catch {} around the assignment.